src/sentry/snuba/metrics.py

```diff
@@ -603,14 +625,16 @@ def _extract_data(self, entity, data, groups):
         )
 
         timestamp = data.pop(TS_COL_GROUP, None)
+        if timestamp is not None:
+            timestamp = parse_snuba_datetime(timestamp)
```

Comment:
No idea how this ever worked without parsing.

Reply:
do you use that timestamp anywhere or is it just passed down to serializer? json serializer converts it back to string (unsure about whether it's the same format)

Reply:
It's used as a key to determine which element in the time series array gets populated by this value:
https://github.com/getsentry/sentry/pull/30087/files#diff-68460020c8d197c96ad96bd390b78594af8d35b799cc2f1372e9f11bd0ecf334L623
IMO its better to use datetimes than serialized strings for this lookup, because there might be subtle differences between the dict keys and how snuba serializes datetimes.